File: 49-group-anagrams/49-group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
         int n = strs.size();
        unordered_map<string, vector<string>> map;
        vector<vector<string>>ans;
        
        int i=0;
        while(i<n){
            string str = strs[i];
            sort(str.begin(), str.end());
            map[str].push_back(strs[i]);
            i++;
        }
        for(auto i=map.begin(); i!= map.end(); i++){
            ans.push_back(i->second);
        }
        return ans;
    }
};

File: 49-group-anagrams/49-group-anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        HashMap<String ,List> map = new HashMap<>();
        if(strs.length==0)return new ArrayList<>();
        
        
        
        for(String str : strs){
            char [] arr = str.toCharArray();
            Arrays.sort(arr);
            String copy_arr = new String(arr);
            
            
            if(!map.containsKey(copy_arr))
                map.put(copy_arr, new ArrayList<>());
            
            map.get(copy_arr).add(str);
        }
        return new ArrayList(map.values());
        
    }
}
  